receiver/mongodbatlasreceiver/mongodb_event_to_logdata.go

```diff
@@ -0,0 +1,166 @@
+)
+
+const (
+	// Number of log attributes to add to the plog.LogRecordSlice.
+	totalLogAttributes = 5
+
+	// Number of resource attributes to add to the plog.ResourceLogs.
+	totalResourceAttributes = 4
+)
+
+// layout for the timestamp format in the plog.Logs structure
+const layout = "2006-01-02T15:04:05.000-07:00"
+
+// Severity mapping of the mongodb atlas logs
+var severityMap = map[string]plog.SeverityNumber{
+	"F":  plog.SeverityNumberFATAL,
+	"E":  plog.SeverityNumberERROR,
+	"W":  plog.SeverityNumberWARN,
+	"I":  plog.SeverityNumberINFO,
+	"D":  plog.SeverityNumberDEBUG,
+	"D1": plog.SeverityNumberDEBUG,
+	"D2": plog.SeverityNumberDEBUG2,
+	"D3": plog.SeverityNumberDEBUG3,
+	"D4": plog.SeverityNumberDEBUG4,
+	"D5": plog.SeverityNumberDEBUG4,
+}
+
+// mongoAuditEventToLogRecord converts model.AuditLog event to plog.LogRecordSlice and adds the resource attributes.
+func mongodbAuditEventToLogData(logger *zap.Logger, e model.AuditLog, pc ProjectContext, hostname, clusterName, logName string) plog.Logs {
+	ld := plog.NewLogs()
+	rl := ld.ResourceLogs().AppendEmpty()
+	sl := rl.ScopeLogs().AppendEmpty()
+	lr := sl.LogRecords().AppendEmpty()
+
+	resourceAttrs := rl.Resource().Attributes()
+	resourceAttrs.EnsureCapacity(totalResourceAttributes)
+
+	// Attributes related to the object causing the event.
+	resourceAttrs.InsertString("mongodb_atlas.org", pc.orgName)
+	resourceAttrs.InsertString("mongodb_atlas.project", pc.Project.Name)
+	resourceAttrs.InsertString("mongodb_atlas.cluster", clusterName)
+	resourceAttrs.InsertString("mongodb_atlas.host.name", hostname)
+
+	data, err := json.Marshal(e)
+	if err != nil {
+		logger.Warn("failed to marshal", zap.Error(err))
+	}
+
+	t, err := time.Parse(layout, e.Timestamp.Date)
+	if err != nil {
+		logger.Warn("Time failed to parse correctly", zap.Error(err))
+	}
+	lr.SetTimestamp(pcommon.NewTimestampFromTime(t))
+	lr.SetObservedTimestamp(pcommon.NewTimestampFromTime(time.Now()))
+
+	// Insert Raw Log message into Body of LogRecord
+	lr.Body().SetStringVal(string(data))
+
+	// Since Audit Logs don't have a severity/level
+	// Set the "SeverityNumber" and "SeverityText" to INFO
+	lr.SetSeverityNumber(plog.SeverityNumberINFO)
+	lr.SetSeverityText("INFO")
+
+	attrs := lr.Attributes()
+	attrs.EnsureCapacity(totalLogAttributes)
+
+	if e.AuthType != "" {
+		attrs.InsertString("authtype", e.AuthType)
+	}
+
+	attrs.InsertString("local.ip", e.Local.IP)
+	attrs.InsertInt("local.port", int64(e.Local.Port))
+	attrs.InsertString("remote.ip", e.Remote.IP)
+	attrs.InsertInt("remote.port", int64(e.Remote.Port))
+	attrs.InsertString("uuid.binary", e.ID.Binary)
+	attrs.InsertString("uuid.type", e.ID.Type)
+	attrs.InsertInt("result", int64(e.Result))
+	attrs.InsertString("log_name", logName)
+
+	if e.Param.User != "" {
+		attrs.InsertString("param.user", e.Param.User)
+		attrs.InsertString("param.database", e.Param.Database)
+		attrs.InsertString("param.mechanism", e.Param.Mechanism)
+	}
+
```

Comment:
```suggestion
	lr := sl.LogRecords().AppendEmpty()
	{
		data, err := json.Marshal(e)
		if err != nil {
			logger.Warn("failed to marshal", zap.Error(err))
		}

		t, err := time.Parse(layout, e.Timestamp.Date)
		if err != nil {
			logger.Warn("Time failed to parse correctly", zap.Error(err))
		}
		lr.SetTimestamp(pcommon.NewTimestampFromTime(t))
		lr.SetObservedTimestamp(pcommon.NewTimestampFromTime(time.Now()))

		// Insert Raw Log message into Body of LogRecord
		lr.Body().SetStringVal(string(data))

		// Since Audit Logs don't have a severity/level
		// Set the "SeverityNumber" and "SeverityText" to INFO
		lr.SetSeverityNumber(plog.SeverityNumberINFO)
		lr.SetSeverityText("INFO")

		attrs := lr.Attributes()
		attrs.EnsureCapacity(totalLogAttributes)

		if e.AuthType != "" {
			attrs.InsertString("authtype", e.AuthType)
		}

		attrs.InsertString("local.ip", e.Local.IP)
		attrs.InsertInt("local.port", int64(e.Local.Port))
		attrs.InsertString("remote.ip", e.Remote.IP)
		attrs.InsertInt("remote.port", int64(e.Remote.Port))
		attrs.InsertString("uuid.binary", e.ID.Binary)
		attrs.InsertString("uuid.type", e.ID.Type)
		attrs.InsertInt("result", int64(e.Result))
		attrs.InsertString("log_name", logName)

		if e.Param.User != "" {
			attrs.InsertString("param.user", e.Param.User)
			attrs.InsertString("param.database", e.Param.Database)
			attrs.InsertString("param.mechanism", e.Param.Mechanism)
		}
	}
```